dev_docs/deep_zoom.md

```diff
@@ -0,0 +1,32 @@
+# Spatial Mode Deep Zoom Feature
+
+The Deep Zoom feature in CELLxGENE Explorer enhances the spatial exploration of datasets by allowing users to zoom in and out of high-resolution images smoothly. This functionality is critical for detailed examination of spatial data, such as those from Slide-seq V2 or Visium, providing fine-grained insight into tissue samples and cellular interactions.
+
+It is built on the OpenSeadragon (OSD) JavaScript library, integrated into the Explorer’s frontend to deliver a responsive zooming experience. The backend handles the generation and storage of deep zoom image tiles, making it efficient to load and view high-resolution spatial data across environments (dev, staging, prod).
+
+## Frontend Integration (Explorer FE)
+
+- [OpenSeadragon (OSD)](https://openseadragon.github.io/): The OSD library is imported into the frontend to handle interactive zoom and pan operations. OSD dynamically loads image tiles based on the user’s zoom level and position.
+- Zoom & Pan Synchronization: Zooming and panning are kept in sync with spatial layers (e.g., dot layers) in the Explorer to ensure smooth interaction across different dataset views.
+- Dynamic Resource Loading: URLs for the image tiles are generated based on the current environment (dev, staging, prod), ensuring the correct assets are fetched dynamically.
+
+## Backend Image Processing & Storage
+
+- CXG Conversion Script: The backend uses a CXG conversion pipeline to process high-resolution spatial images and generate the necessary image tiles and manifest files (DZI format) for deep zoom.
+- Storage on S3: The generated deep zoom assets (tiles and DZI files) are uploaded to a dedicated Amazon S3 bucket (spatial-deep-zoom). Each dataset has its own folder in the bucket (e.g., spatial-deep-zoom/{DATASET_VERSION_ID}).
```

Comment:
discussion: Should we emphasize that it is `DATASET_VERSION_ID` instead of `DATASET_ID` because `DATASET_ID` doesn't take the same dataset's revisions into account, but the `DATASET_VERSION_ID` does. Thus us using `DATASET_VERSION_ID` in the directory name in `spatial-deep-zoom/{DATASET_VERSION_ID}`?